packages/drawer/src/ux-drawer.ts

```diff
@@ -0,0 +1,302 @@
+  private viewportType: 'mobile' | 'desktop' = 'desktop';
+  private overlayElement: HTMLElement;
+  private contentWrapperElement: HTMLElement;
+  private contentElement: HTMLElement;
+  private showed: boolean = false;
+  private showing: boolean = false;
+  private hiding: boolean = false;
+
+  constructor(
+    public element: HTMLElement,
+    private styleEngine: StyleEngine,
+    private modalService: ModalService,
+    private taskQueue: TaskQueue,
+    private defaultConfig: DefaultDrawerConfiguration) {
+      if (this.defaultConfig.modalBreakpoint !== undefined) {
+        this.modalBreakpoint = this.defaultConfig.modalBreakpoint
+      }
+      if (this.defaultConfig.host !== undefined) {
+        this.host = this.defaultConfig.host
+      }
+      if (this.defaultConfig.overlayDismiss !== undefined) {
+        this.overlayDismiss = this.defaultConfig.overlayDismiss
+      }
+      if (this.defaultConfig.position !== undefined) {
+        this.position = this.defaultConfig.position
+      }
+      if (this.defaultConfig.keyboard !== undefined) {
+        this.keyboard = this.defaultConfig.keyboard
+      }
+      if (this.defaultConfig.theme !== undefined) {
+        this.theme = this.defaultConfig.theme
+      }
+    }
+  private bindingContext: any;
+  public bind(bindingContext: any) {
+    this.bindingContext = bindingContext;
+    this.themeChanged(this.theme);
+    this.setViewportType();
+    window.addEventListener('resize', this);
+    this.positionChanged();
+    this.modalBreakpointChanged();
+    this.hostChanged();
+    this.overlayDismissChanged();
+    this.keyboardChanged();
+  }
+
+  public positionChanged() {
+    if (!this.position && this.defaultConfig.position) {
+      this.position = this.defaultConfig.position;
+    }
+  }
+
+  public modalBreakpointChanged() {
+    if (typeof this.modalBreakpoint !== 'number' && this.defaultConfig.modalBreakpoint) {
+      this.modalBreakpoint = this.defaultConfig.modalBreakpoint;
+    }
+  }
+
+  public hostChanged() {
+    if (this.host === false || this.host === 'body' || this.host instanceof HTMLElement) {
+      return;
+    }
+    if (this.defaultConfig.host !== undefined) {
+      this.host = this.defaultConfig.host;
+      return;
+    }
+    if (this.host === '') {
+      this.host = 'body';
+    }
+  }
+
+  public overlayDismissChanged() {
+    if (!this.overlayDismiss && this.defaultConfig.overlayDismiss) {
+      this.overlayDismiss = this.defaultConfig.overlayDismiss;
+    }
+  }
+
+  public keyboardChanged() {
+    if (!this.keyboard && this.defaultConfig.keyboard) {
+      this.keyboard = this.defaultConfig.keyboard;
+    }
+  }
+
+  public attached() {
+    if (this.host) {
+      this.moveToHost();
+    }
+    this.show();
+  }
+
+  public detached() {
+    if (this.host) {
+      this.removeFromHost();
+    }
+  }
+
+  private show() {
+    if (this.showing && this.showed) {return;}
+    if(DOM.activeElement instanceof HTMLElement) {
```

Comment:
if we are to use `DOM`, better use it everywhere, else we can use `document.activeElement` instead